commons/all/src/test/java/org/infinispan/commons/util/TypedPropertiesTest.java

```diff
@@ -0,0 +1,89 @@
+package org.infinispan.commons.util;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Properties;
+
+import org.junit.Test;
+
+/**
+ * Test for {@link TypedProperties}.
+ *
+ * @author Diego Lovison
+ * @since 12.1
+ **/
+public class TypedPropertiesTest {
+
+   @Test
+   public void testIntProperty() {
+      TypedProperties p = createProperties();
+      assertEquals(1, p.getIntProperty("int", 999));
+      assertEquals(10, p.getIntProperty("int_put_str", 999));
+      assertEquals(1, p.getIntProperty("int_invalid", 1));
+      assertEquals(1, p.getIntProperty("int_null", 1));
+
+      System.setProperty("fooVar", "1");
+      assertEquals(1, p.getIntProperty("int_key_value_replacement", 1, true));
```

Comment:
`fooVar` is `1` and the `defaultValue` is `1`. you are not sure if the `1` returned is because the replacement happened or if it is the `defaultValue`

Reply:
Fixed